lib/objFile.ml

```diff
@@ -0,0 +1,229 @@
+(************************************************************************)
+(*         *   The Coq Proof Assistant / The Coq Development Team       *)
+(*  v      *         Copyright INRIA, CNRS and contributors             *)
+(* <O___,, * (see version control and CREDITS file for authors & dates) *)
+(*   \VV/  **************************************************************)
+(*    //   *    This file is distributed under the terms of the         *)
+(*         *     GNU Lesser General Public License Version 2.1          *)
+(*         *     (see LICENSE file for the text of the license)         *)
+(************************************************************************)
+
+open Pp
+open System
+
+let magic_number = 0x436F7121l (* "Coq!" *)
```

Comment:
where does this come from?

Reply:
As far as I know there is no "registry" of magic numbers you could apply to. I'm quite sympathetic to a stackoverflow message that suggests you could append to the magic number an ASCII string saying "This is Coq binary file version/format\n" followed by binary data.

Reply:
> where does this come from?

This was specified in the CEP.

> you could append to the magic number an ASCII string

I am not sure I understand what you want. Do you mean that you want the magic number to be the literal string "This is Coq [...]"? This seems a bit exaggerated, traditionally magic numbers are 4 bytes wide, and despite the lack of a central authority, there is quite a bit of root in there to prevent clashing.

Reply:
ginven that `file` is not going to know the magic number, I thought that having after the magic number a readable string could help (as in `head -1 file` would tell me enough).

I know it was in the CEP, I was asking why that number specifically.

Reply:
If we decide to set this magic number in stone, we can write a patch for `file` to support `Coq` object files. I chose that number because it's 4 bytes wide, starts with "Coq", and I added a random printable character at the end that made it look like a sentence.